## Q1) A company started to invest in digital marketing as a new way of their product promotions. For that they collected data and decided to carry out a study on it.
## * The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.
## * The company needs to check whether there is any dependency between the features “Region” and “Manager”.
## Help the company to carry out their study with the help of data provided.

In [78]:
# importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [79]:
# Load a comma separated file (CSV file) into a DataFrame: 
data = pd.read_csv('C:/Users/Lenovo/Assignment/assignment 4/DATA/Sales_add.csv')

In [80]:
# displaying the first 5 rows of the dataframe by default
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [81]:
# checking null values in the dataframe
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


^ 22 entries are there, and each columns are having 22 non-null values, hence no null value present in the dataframe

### ~ The company wishes to clarify whether there is any increase in sales after stepping into digital marketing.

#### Hypothesis
* Ho: on increase in sales after stepping into digital marketing
* HA: increase in sales after stepping into digital marketing
* taking alpha = 0.05

^ Using 2 sample t test, since we have to determine whether digital marketing has an effect on the sales or not, sample size is very small (sample size -22) and we have 2 column values that has to be used in the test.

In [82]:
# using the ttest_ind() function from the scipy.stats library to conduct a two sample t-test
from scipy.stats import ttest_ind

In [83]:
# store the values of the column 'Sales_before_digital_add(in $)' in array format into a new variable
before_sales = data['Sales_before_digital_add(in $)'].values

In [84]:
# display values stored in 'before_sales'
before_sales

array([132921, 149559, 146278, 152167, 159525, 137163, 130625, 131140,
       171259, 141956, 159339, 178939, 145062, 151514, 147463, 177195,
       140862, 167996, 132135, 152493, 147425, 130263], dtype=int64)

In [85]:
# store the values of the column 'Sales_After_digital_add(in $)' in array format into a new variable
after_sales = data['Sales_After_digital_add(in $)'].values

In [86]:
# display values stored in 'after_sales'
after_sales

array([270390, 223334, 244243, 231808, 258402, 256948, 222106, 230637,
       226261, 193735, 203175, 276279, 205180, 253131, 229336, 187305,
       234908, 191517, 227040, 212579, 263388, 243020], dtype=int64)

In [87]:
# find variance for before_sales and after_sales
print(np.var(before_sales), np.var(after_sales))


210329900.67975205 623460269.4710743


In [88]:
# The ratio of the larger sample variance to the smaller sample variance
623460269.4710743/210329900.67975205

2.964201796587894

^ The ratio of the larger sample variance to the smaller sample variance  = 2.964201796587894, which is less than 4. This means we can assume that the population variances are equal.

In [89]:
# performing 2 sample t test using the values stored in 'before_sales' and 'after_sales'
# storing t test statistic and  corresponding  p-value in 't_test' and 'p_value'
t_test, p_value=ttest_ind(before_sales, after_sales)

In [90]:
# display value stored in 't_test'
t_test

-12.995084451110875

In [91]:
# display value stored in 'p_value'
p_value

2.614368006904645e-16

* The t test statistic is -12.995084451110875 and the corresponding  p-value is 2.614368006904645e-16

In [92]:
# if p<alpha - reject null hypothesis
if p_value<0.05:
    print('Null Hypothesis Rejected')
else:
    print('Failed to reject Null Hypothesis')
    

Null Hypothesis Rejected


* HA: increase in sales after stepping into digital marketing

^ The above output gives 'Null Hypothesis Rejected', so accepting Alternate Hypothesis which indicates that there is an increase in  sales after stepping into digital marketing. 


### ~The company needs to check whether there is any dependency between the features “Region” and “Manager”.

#### Hypothesis
* Ho:No dependency between the features “Region” and “Manager”
* HA: dependency between the features “Region” and “Manager”
* taking alpha = 0.05

^ using Chi-Square Test of Independence since we have to determine whether or not there is a significant association between two categorical variables [“Region” and “Manager”]

In [93]:
# used to find the count each regions having the same manager
# grouped the dataframe by 'Region'and 'Manager and count the number of rows coming under the grouped 
# dataframe and unstacking by 'Manager' column
data1 = data.groupby(['Region','Manager']).size().unstack('Manager')
data1

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

In [94]:
# for importing chi2_contingency function from the scipy library
from scipy.stats import chi2_contingency

In [95]:
# storing the values from dataframe 'data1' to 'tab-data'
tab_data = [[4, 3, 3], [4, 1, 2], [1, 3, 1]]

In [96]:
# performing Chi-Square Test of Independence using the values stored in 'tab_data'
# storing Chi-Square Test Statistic,  p-value, Degrees of freedom,  expected values for each cell in 
# the contingency table
c, p, df, expected=chi2_contingency(tab_data)

In [97]:
# used for storing Chi-Square Test Statistic
c

3.050566893424036

In [98]:
# used for storing p-value:
p

0.5493991051158094

In [99]:
# used for storing Degrees of freedom
df

4

In [100]:
# used for storing expected values for each cell in the contingency table
expected

array([[4.09090909, 3.18181818, 2.72727273],
       [2.86363636, 2.22727273, 1.90909091],
       [2.04545455, 1.59090909, 1.36363636]])

* Chi-Square Test Statistic: 3.050566893424036
* p-value: 0.5493991051158094
* Degrees of freedom: 4 
* Array: displays the expected values for each cell in the contingency table.
array([[4.09090909, 3.18181818, 2.72727273],
       [2.86363636, 2.22727273, 1.90909091],
       [2.04545455, 1.59090909, 1.36363636]])

In [101]:
# if p<alpha - reject null hypothesis
if p<0.05:
    print('Null Hypothesis Rejected')
else:
    print('Failed to reject Null Hypothesis')

Failed to reject Null Hypothesis


* Ho:No dependency between the features “Region” and “Manager”

^  The above output gives 'Failed to reject Null Hypothesis', hence null hypothesis accepted which indicates that there is no dependency between the features “Region” and “Manager”